File: apps/mobile/android/app/src/main/java/com/tchat/screens/MoreScreen.kt
package com.tchat.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tchat.designsystem.Colors
import com.tchat.designsystem.Spacing

/**
 * Settings and additional features screen
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MoreScreen() {
    var showingProfile by remember { mutableStateOf(false) }
    var darkModeEnabled by remember { mutableStateOf(false) }
    var notificationsEnabled by remember { mutableStateOf(true) }
    var soundEnabled by remember { mutableStateOf(true) }

    // Menu sections
    val menuSections = listOf(
        MoreSection(
            title = "Account",
            items = listOf(
                MoreItem("Profile", Icons.Default.Person, MoreItemType.PROFILE),
                MoreItem("Privacy", Icons.Default.Security, MoreItemType.PRIVACY),
                MoreItem("Security", Icons.Default.VerifiedUser, MoreItemType.SECURITY),
                MoreItem("Billing", Icons.Default.CreditCard, MoreItemType.BILLING)
            )
        ),
        MoreSection(
            title = "Preferences",
            items = listOf(
                MoreItem("Notifications", Icons.Default.Notifications, MoreItemType.NOTIFICATIONS),
                MoreItem("Appearance", Icons.Default.Palette, MoreItemType.APPEARANCE),
                MoreItem("Language", Icons.Default.Language, MoreItemType.LANGUAGE),
                MoreItem("Storage", Icons.Default.Storage, MoreItemType.STORAGE)
            )
        ),
        MoreSection(
            title = "Support",
            items = listOf(
                MoreItem("Help Center", Icons.Default.Help, MoreItemType.HELP),
                MoreItem("Contact Us", Icons.Default.Email, MoreItemType.CONTACT),
                MoreItem("Report Issue", Icons.Default.Report, MoreItemType.REPORT),
                MoreItem("About", Icons.Default.Info, MoreItemType.ABOUT)
            )
        )
    )

    LazyColumn(
        modifier = Modifier
            .fillMaxSize()
            .background(Colors.background),
        contentPadding = PaddingValues(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.lg)
    ) {
        item {
            // Top app bar space
            Text(
                text = "More",
                fontSize = 24.sp,
                fontWeight = FontWeight.Bold,
                color = Colors.textPrimary,
                modifier = Modifier.padding(vertical = Spacing.sm)
            )
        }

        item {
            // Profile header
            ProfileHeaderCard(onProfileClick = { showingProfile = true })
        }

        items(menuSections) { section ->
            MenuSectionCard(
                section = section,
                darkModeEnabled = darkModeEnabled,
                notificationsEnabled = notificationsEnabled,
                onDarkModeToggle = { darkModeEnabled = !darkModeEnabled },
                onNotificationsToggle = { notificationsEnabled = !notificationsEnabled }
            )
        }

        item {
            // Sign out button
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable { /* Sign out */ },
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(
                    containerColor = Colors.surface
                ),
                elevation = CardDefaults.cardElevation(
                    defaultElevation = 4.dp
                )
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(Spacing.md),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.ExitToApp,
                        contentDescription = "Sign Out",
                        tint = Colors.error,
                        modifier = Modifier.size(24.dp)
                    )
                    Spacer(modifier = Modifier.width(Spacing.md))
                    Text(
                        text = "Sign Out",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Medium,
                        color = Colors.error
                    )
                }
            }
        }

        item {
            // Version info
            Text(
                text = "Version 1.0.0 (Build 1)",
                fontSize = 12.sp,
                color = Colors.textSecondary,
                modifier = Modifier
                    .fillMaxWidth()
                    .wrapContentWidth(Alignment.CenterHorizontally)
                    .padding(bottom = Spacing.xl)
            )
        }
    }
}

// MARK: - Data Classes
data class MoreSection(
    val title: String,
    val items: List<MoreItem>
)

data class MoreItem(
    val title: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector,
    val type: MoreItemType
)

enum class MoreItemType {
    PROFILE, PRIVACY, SECURITY, BILLING,
    NOTIFICATIONS, APPEARANCE, LANGUAGE, STORAGE,
    HELP, CONTACT, REPORT, ABOUT
}

// MARK: - Profile Header Card
@Composable
private fun ProfileHeaderCard(onProfileClick: () -> Unit) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable { onProfileClick() },
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = androidx.compose.ui.graphics.Color.White
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.md),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Profile avatar
            Box(
                modifier = Modifier
                    .size(60.dp)
                    .background(
                        color = Colors.primary.copy(alpha = 0.2f),
                        shape = CircleShape
                    ),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "JD",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Colors.primary
                )
            }

            Spacer(modifier = Modifier.width(Spacing.md))

            // Profile info
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "John Doe",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Colors.textPrimary
                )
                Text(
                    text = "john.doe@example.com",
                    fontSize = 14.sp,
                    color = Colors.textSecondary
                )
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(
                        modifier = Modifier
                            .size(8.dp)
                            .background(
                                color = Colors.success,
                                shape = CircleShape
                            )
                    )
                    Spacer(modifier = Modifier.width(Spacing.xs))
                    Text(
                        text = "Online",
                        fontSize = 12.sp,
                        color = Colors.success
                    )
                }
            }

            Icon(
                imageVector = Icons.Default.ChevronRight,
                contentDescription = "View profile",
                tint = Colors.textSecondary
            )
        }
    }
}

// MARK: - Menu Section Card
@Composable
private fun MenuSectionCard(
    section: MoreSection,
    darkModeEnabled: Boolean,
    notificationsEnabled: Boolean,
    onDarkModeToggle: () -> Unit,
    onNotificationsToggle: () -> Unit
) {
    Column {
        Text(
            text = section.title,
            fontSize = 16.sp,
            fontWeight = FontWeight.SemiBold,
            color = Colors.textPrimary,
            modifier = Modifier.padding(bottom = Spacing.sm)
        )

        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(12.dp),
            colors = CardDefaults.cardColors(
                containerColor = androidx.compose.ui.graphics.Color.White
            ),
            elevation = CardDefaults.cardElevation(
                defaultElevation = 4.dp
            )
        ) {
            Column {
                section.items.forEachIndexed { index, item ->
                    MenuItemRow(
                        item = item,
                        isLast = index == section.items.size - 1,
                        darkModeEnabled = darkModeEnabled,
                        notificationsEnabled = notificationsEnabled,
                        onDarkModeToggle = onDarkModeToggle,
                        onNotificationsToggle = onNotificationsToggle
                    )
                }
            }
        }
    }
}

// MARK: - Menu Item Row
@Composable
private fun MenuItemRow(
    item: MoreItem,
    isLast: Boolean,
    darkModeEnabled: Boolean,
    notificationsEnabled: Boolean,
    onDarkModeToggle: () -> Unit,
    onNotificationsToggle: () -> Unit
) {
    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable {
                    when (item.type) {
                        MoreItemType.APPEARANCE -> onDarkModeToggle()
                        MoreItemType.NOTIFICATIONS -> onNotificationsToggle()
                        else -> {
                            // Handle other item types
                            println("Clicked: ${item.title}")
                        }
                    }
                }
                .padding(Spacing.md),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Icon
            Icon(
                imageVector = item.icon,
                contentDescription = item.title,
                tint = getIconColor(item.type),
                modifier = Modifier.size(24.dp)
            )

            Spacer(modifier = Modifier.width(Spacing.md))

            // Title
            Text(
                text = item.title,
                fontSize = 16.sp,
                color = Colors.textPrimary,
                modifier = Modifier.weight(1f)
            )

            // Trailing element (toggle or chevron)
            when (item.type) {
                MoreItemType.NOTIFICATIONS -> {
                    Switch(
                        checked = notificationsEnabled,
                        onCheckedChange = { onNotificationsToggle() },
                        colors = SwitchDefaults.colors(
                            checkedThumbColor = Colors.textOnPrimary,
                            checkedTrackColor = Colors.primary
                        )
                    )
                }
                MoreItemType.APPEARANCE -> {
                    Switch(
                        checked = darkModeEnabled,
                        onCheckedChange = { onDarkModeToggle() },
                        colors = SwitchDefaults.colors(
                            checkedThumbColor = Colors.textOnPrimary,
                            checkedTrackColor = Colors.primary
                        )
                    )
                }
                else -> {
                    Icon(
                        imageVector = Icons.Default.ChevronRight,
                        contentDescription = "Navigate",
                        tint = Colors.textSecondary,
                        modifier = Modifier.size(16.dp)
                    )
                }
            }
        }

        if (!isLast) {
            Divider(
                modifier = Modifier.padding(start = 56.dp),
                color = Colors.border,
                thickness = 0.5.dp
            )
        }
    }
}

@Composable
private fun getIconColor(type: MoreItemType): androidx.compose.ui.graphics.Color {
    return when (type) {
        MoreItemType.REPORT -> Colors.warning
        MoreItemType.SECURITY, MoreItemType.PRIVACY -> Colors.success
        MoreItemType.BILLING -> Colors.primary
        else -> Colors.textSecondary
    }
}

// MARK: - Preview
@Preview(showBackground = true)
@Composable
fun MoreScreenPreview() {
    MoreScreen()
}

File: apps/mobile/ios/Sources/Screens/MoreScreen.swift
//
//  MoreScreen.swift
//  TchatApp
//
//  Created by Claude on 22/09/2024.
//

import SwiftUI

/// Settings and additional features screen
public struct MoreScreen: View {
    @State private var showingProfile = false
    @State private var darkModeEnabled = false
    @State private var notificationsEnabled = true
    @State private var soundEnabled = true

    private let colors = Colors()
    private let spacing = Spacing()

    // Menu sections
    private let menuSections = [
        MoreSection(
            title: "Account",
            items: [
                MoreItem("Profile", "person.circle.fill", .profile),
                MoreItem("Privacy", "lock.shield.fill", .privacy),
                MoreItem("Security", "checkmark.shield.fill", .security),
                MoreItem("Billing", "creditcard.fill", .billing)
            ]
        ),
        MoreSection(
            title: "Preferences",
            items: [
                MoreItem("Notifications", "bell.fill", .notifications),
                MoreItem("Appearance", "paintbrush.fill", .appearance),
                MoreItem("Language", "globe", .language),
                MoreItem("Storage", "internaldrive.fill", .storage)
            ]
        ),
        MoreSection(
            title: "Support",
            items: [
                MoreItem("Help Center", "questionmark.circle.fill", .help),
                MoreItem("Contact Us", "envelope.fill", .contact),
                MoreItem("Report Issue", "exclamationmark.triangle.fill", .report),
                MoreItem("About", "info.circle.fill", .about)
            ]
        )
    ]

    public init() {}

    public var body: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: spacing.lg) {
                    // Profile header
                    ProfileHeaderView(showingProfile: $showingProfile)

                    // Menu sections
                    ForEach(Array(menuSections.enumerated()), id: \.offset) { index, section in
                        MenuSectionView(
                            section: section,
                            darkModeEnabled: $darkModeEnabled,
                            notificationsEnabled: $notificationsEnabled,
                            soundEnabled: $soundEnabled
                        )
                    }

                    // Sign out button
                    Button(action: {
                        // Sign out action
                    }) {
                        HStack {
                            Image(systemName: "rectangle.portrait.and.arrow.right")
                                .foregroundColor(colors.error)
                            Text("Sign Out")
                                .font(.system(size: 16, weight: .medium))
                                .foregroundColor(colors.error)
                        }
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, spacing.md)
                        .background(colors.surface)
                        .cornerRadius(12)
                    }
                    .padding(.horizontal, spacing.md)

                    // Version info
                    Text("Version 1.0.0 (Build 1)")
                        .font(.system(size: 12))
                        .foregroundColor(colors.textSecondary)
                        .padding(.bottom, spacing.xl)
                }
                .padding(.top, spacing.sm)
            }
            .navigationTitle("More")
            .navigationBarTitleDisplayMode(.large)
        }
        .navigationViewStyle(StackNavigationViewStyle())
        .sheet(isPresented: $showingProfile) {
            ProfileView()
        }
    }
}

// MARK: - Data Models
private struct MoreSection {
    let title: String
    let items: [MoreItem]
}

private struct MoreItem {
    let title: String
    let icon: String
    let type: MoreItemType

    init(_ title: String, _ icon: String, _ type: MoreItemType) {
        self.title = title
        self.icon = icon
        self.type = type
    }
}

private enum MoreItemType {
    case profile, privacy, security, billing
    case notifications, appearance, language, storage
    case help, contact, report, about
}

// MARK: - Profile Header View
private struct ProfileHeaderView: View {
    @Binding var showingProfile: Bool

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        Button(action: {
            showingProfile = true
        }) {
            HStack(spacing: spacing.md) {
                // Profile avatar
                Circle()
                    .fill(colors.primary.opacity(0.2))
                    .frame(width: 60, height: 60)
                    .overlay(
                        Text("JD")
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(colors.primary)
                    )

                // Profile info
                VStack(alignment: .leading, spacing: spacing.xs) {
                    Text("John Doe")
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(colors.textPrimary)

                    Text("john.doe@example.com")
                        .font(.system(size: 14))
                        .foregroundColor(colors.textSecondary)

                    HStack(spacing: spacing.xs) {
                        Circle()
                            .fill(colors.success)
                            .frame(width: 8, height: 8)
                        Text("Online")
                            .font(.system(size: 12))
                            .foregroundColor(colors.success)
                    }
                }

                Spacer()

                Image(systemName: "chevron.right")
                    .foregroundColor(colors.textSecondary)
            }
            .padding(spacing.md)
            .background(Color.white)
            .cornerRadius(12)
            .shadow(color: colors.shadowLight, radius: 4, y: 2)
        }
        .padding(.horizontal, spacing.md)
    }
}

// MARK: - Menu Section View
private struct MenuSectionView: View {
    let section: MoreSection
    @Binding var darkModeEnabled: Bool
    @Binding var notificationsEnabled: Bool
    @Binding var soundEnabled: Bool

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        VStack(alignment: .leading, spacing: spacing.sm) {
            // Section title
            Text(section.title)
                .font(.system(size: 16, weight: .semibold))
                .foregroundColor(colors.textPrimary)
                .padding(.horizontal, spacing.md)

            // Section items
            VStack(spacing: 0) {
                ForEach(Array(section.items.enumerated()), id: \.offset) { index, item in
                    MenuItemView(
                        item: item,
                        isLast: index == section.items.count - 1,
                        darkModeEnabled: $darkModeEnabled,
                        notificationsEnabled: $notificationsEnabled,
                        soundEnabled: $soundEnabled
                    )
                }
            }
            .background(Color.white)
            .cornerRadius(12)
            .shadow(color: colors.shadowLight, radius: 4, y: 2)
            .padding(.horizontal, spacing.md)
        }
    }
}

// MARK: - Menu Item View
private struct MenuItemView: View {
    let item: MoreItem
    let isLast: Bool
    @Binding var darkModeEnabled: Bool
    @Binding var notificationsEnabled: Bool
    @Binding var soundEnabled: Bool

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        Button(action: {
            handleItemTap(item.type)
        }) {
            HStack(spacing: spacing.md) {
                // Icon
                Image(systemName: item.icon)
                    .font(.system(size: 18))
                    .foregroundColor(iconColor(for: item.type))
                    .frame(width: 24, height: 24)

                // Title
                Text(item.title)
                    .font(.system(size: 16))
                    .foregroundColor(colors.textPrimary)

                Spacer()

                // Trailing element (toggle or chevron)
                trailingElement(for: item.type)
            }
            .padding(.horizontal, spacing.md)
            .padding(.vertical, spacing.md)
        }
        .buttonStyle(PlainButtonStyle())

        if !isLast {
            Divider()
                .padding(.leading, 56)
        }
    }

    @ViewBuilder
    private func trailingElement(for type: MoreItemType) -> some View {
        switch type {
        case .notifications:
            Toggle("", isOn: $notificationsEnabled)
                .toggleStyle(SwitchToggleStyle(tint: colors.primary))
        case .appearance:
            Toggle("", isOn: $darkModeEnabled)
                .toggleStyle(SwitchToggleStyle(tint: colors.primary))
        default:
            Image(systemName: "chevron.right")
                .font(.system(size: 14))
                .foregroundColor(colors.textSecondary)
        }
    }

    private func iconColor(for type: MoreItemType) -> Color {
        switch type {
        case .report:
            return colors.warning
        case .security, .privacy:
            return colors.success
        case .billing:
            return colors.primary
        default:
            return colors.textSecondary
        }
    }

    private func handleItemTap(_ type: MoreItemType) {
        // Handle different item types
        switch type {
        case .profile:
            print("Open profile")
        case .privacy:
            print("Open privacy settings")
        case .security:
            print("Open security settings")
        case .billing:
            print("Open billing")
        case .language:
            print("Open language settings")
        case .storage:
            print("Open storage settings")
        case .help:
            print("Open help center")
        case .contact:
            print("Open contact")
        case .report:
            print("Open report issue")
        case .about:
            print("Open about")
        default:
            break
        }
    }
}

// MARK: - Profile View (Placeholder)
private struct ProfileView: View {
    @Environment(\.presentationMode) var presentationMode

    private let colors = Colors()

    var body: some View {
        NavigationView {
            VStack {
                Text("Profile Details")
                    .font(.title2)
                    .foregroundColor(colors.textPrimary)

                Spacer()

                Button("Save") {
                    presentationMode.wrappedValue.dismiss()
                }
                .foregroundColor(colors.primary)
            }
            .navigationTitle("Profile")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    Button("Cancel") {
                        presentationMode.wrappedValue.dismiss()
                    }
                }
            }
        }
    }
}

// MARK: - Preview
#if DEBUG
struct MoreScreen_Previews: PreviewProvider {
    static var previews: some View {
        MoreScreen()
            .previewDisplayName("More Screen")
    }
}
#endif